src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -660,41 +706,84 @@ private void initializeInputPreloadExecutorService() {
+        partitionBuilder.setBegin(begin);
+        partitionBuilder.setEnd(end);
+        partitionBuilder.setWorkspace(workspace);
+        partitionBuilder.setGenerateArrayNameFromPartitionBounds(true);
+        return partitionBuilder.build();
+    }
+
+    private List<GenomicsDBImportConfiguration.Partition> generatePartitionListFromWorkspace(final String workspace) {
+        String[] partitions = GenomicsDBUtils.listGenomicsDBArrays(workspace);
+        List<GenomicsDBImportConfiguration.Partition> configPartitions = new ArrayList<>();
+        for (int i=0; i<partitions.length; i++) {
+            long[] bounds = GenomicsDBUtils.getArrayColumnBounds(workspace, partitions[i]);
+            Coordinates.GenomicsDBColumn.Builder beginBuilder = Coordinates.GenomicsDBColumn.newBuilder();
+            Coordinates.GenomicsDBColumn.Builder endBuilder = Coordinates.GenomicsDBColumn.newBuilder();
+            beginBuilder.setTiledbColumn(bounds[0]);
+            endBuilder.setTiledbColumn(bounds[1]);
+            configPartitions.add(createPartitionWithBeginAndEnd(beginBuilder.build(), endBuilder.build()));
+        }
+        return configPartitions;
+    }
+
     private List<GenomicsDBImportConfiguration.Partition> generatePartitionListFromIntervals(List<SimpleInterval> chromosomeIntervals) {
         return chromosomeIntervals.stream().map(interval -> {
-            GenomicsDBImportConfiguration.Partition.Builder partitionBuilder = GenomicsDBImportConfiguration.Partition.newBuilder();
             Coordinates.ContigPosition.Builder contigPositionBuilder = Coordinates.ContigPosition.newBuilder();
-            Coordinates.GenomicsDBColumn.Builder columnBuilder = Coordinates.GenomicsDBColumn.newBuilder();
+            Coordinates.GenomicsDBColumn.Builder beginBuilder = Coordinates.GenomicsDBColumn.newBuilder();
+            Coordinates.GenomicsDBColumn.Builder endBuilder = Coordinates.GenomicsDBColumn.newBuilder();
             //begin
             contigPositionBuilder.setContig(interval.getContig()).setPosition(interval.getStart());
-            columnBuilder.setContigPosition(contigPositionBuilder.build());
-            partitionBuilder.setBegin(columnBuilder.build());
+            beginBuilder.setContigPosition(contigPositionBuilder.build());
             //end
             contigPositionBuilder.setPosition(interval.getEnd());
-            columnBuilder.setContigPosition(contigPositionBuilder.build());
-            partitionBuilder.setEnd(columnBuilder.build());
-            partitionBuilder.setWorkspace(workspace);
-            partitionBuilder.setGenerateArrayNameFromPartitionBounds(true);
-            return partitionBuilder.build();
+            endBuilder.setContigPosition(contigPositionBuilder.build());
+            return createPartitionWithBeginAndEnd(beginBuilder.build(), endBuilder.build());
         }).collect(Collectors.toList());
     }
 
+    private void generateIntervalListFromVidMap() {
```

Comment:
I would make these less stateful by returning the list of intervals instead of modifying it in place.

Reply:
done